frontend/src/Components/ScaleSelector/ScaleSelector.tsx

```diff
@@ -7,122 +8,143 @@ import classes from './ScaleSelector.module.css';
 
 const availableScales = Object.entries(SCALES);
 
-export const ScaleSelector = connectToWebSocket(({ socket: { connected, setScale } }) => {
-  const [open, setOpen] = useState(false);
-  const [selected, setSelected] = useState(-1);
-  const selectionButtonRef = useRef<HTMLButtonElement>(null);
-  const dropdownRef = useRef<HTMLUListElement>(null);
+export const ScaleSelector = connectToWebSocket(
+  ({
+    socket: {
+      connected,
+      setScale,
+      state: { scale },
+    },
+  }) => {
+    const [open, setOpen] = useState(false);
+    const [selected, setSelected] = useState(-1);
+    const selectionButtonRef = useRef<HTMLButtonElement>(null);
+    const dropdownRef = useRef<HTMLUListElement>(null);
 
-  const close = () => {
-    setOpen(false);
-    setSelected(-1);
-  };
+    const close = () => {
+      setOpen(false);
+      setSelected(-1);
+    };
 
-  const toggle = () => {
-    if (open) {
-      close();
-    } else {
-      setSelected(0);
-      setOpen(true);
-    }
-  };
-
-  const handleBlur = ({ relatedTarget }: FocusEvent) => {
-    if (relatedTarget !== selectionButtonRef.current && relatedTarget !== dropdownRef.current) {
-      close();
-    }
-  };
-
-  const handleKeyboardNavigation = (event: KeyboardEvent) => {
-    switch (event.key) {
-      case 'ArrowDown':
-        setSelected((selected + 1) % availableScales.length);
-        event.preventDefault();
-        break;
-      case 'ArrowUp':
-        setSelected((Math.max(selected, 0) + availableScales.length - 1) % availableScales.length);
-        event.preventDefault();
-        break;
-      case 'Home':
-      case 'PageUp':
-        setSelected(0);
-        break;
-      case 'End':
-      case 'PageDown':
-        setSelected(availableScales.length - 1);
-        break;
-      case 'Escape':
+    const toggle = () => {
+      if (open) {
         close();
-        selectionButtonRef.current?.focus();
-        break;
-      case 'Enter':
-      case ' ':
-        // Prevent "enter" from triggering the button if the dropdown is open
-        event.preventDefault();
+      } else {
+        setSelected(availableScales.findIndex(isScaleSelected(scale)) || 0);
```

Comment:
What is the point of `|| 0`? Because `findIndex` returns `-1` if it does not find the index, the only possible falsy result is `0`. Since there is an explicit test for the not found case that tests for `-1`, I would assume that not selecting anything is the desired case here.

```suggestion
        setSelected(availableScales.findIndex(isScaleSelected(scale)));
```

Reply:
Ah you are right. Somehow I assumed it would be undefined, but of course that does not make sense.